frontend/packages/console-app/src/components/nodes/NodeLogs.tsx

```diff
@@ -0,0 +1,298 @@
+import * as React from 'react';
+import {
+  Alert,
+  Checkbox,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateVariant,
+  Select,
+  SelectOption,
+  Title,
+} from '@patternfly/react-core';
+import classnames from 'classnames';
+import { Trans, useTranslation } from 'react-i18next';
+import { coFetch } from '@console/internal/co-fetch';
+import {
+  ExternalLink,
+  getQueryArgument,
+  LoadingBox,
+  LoadingInline,
+  removeQueryArgument,
+  setQueryArgument,
+} from '@console/internal/components/utils';
+import { modelFor, NodeKind, resourceURL } from '@console/internal/module/k8s';
+import { useUserSettings } from '@console/shared';
+import { USERSETTINGS_PREFIX } from '@console/shared/src/constants';
+import NodeLogsFilterUnit from './NodeLogsUnitFilter';
+import './node-logs.scss';
+
+type NodeLogsProps = {
+  obj: NodeKind;
+};
+
+const NodeLogs: React.FC<NodeLogsProps> = ({ obj: node }) => {
+  const {
+    kind,
+    metadata: { labels, name, namespace: ns },
+    status,
+  } = node;
+  const isWindows = status?.nodeInfo.operatingSystem === 'windows';
+  const pathItems = ['journal'];
+  isWindows
+    ? pathItems.push('containers', 'hybrid-overlay', 'kube-proxy', 'kubelet')
+    : labels['node-role.kubernetes.io/master'] === '' &&
+      pathItems.push('openshift-apiserver', 'kube-apiserver', 'oauth-apiserver');
+  const pathQueryArgument = 'path';
+  const unitQueryArgument = 'unit';
+  const logQueryArgument = 'log';
+
+  const [path, setPath] = React.useState(getQueryArgument(pathQueryArgument) || pathItems[0]);
+  const [logURL, setLogURL] = React.useState('');
+  const [logFilenames, setLogFilenames] = React.useState([]);
+  const [unit, setUnit] = React.useState(getQueryArgument(unitQueryArgument));
+  const [logFilename, setLogFilename] = React.useState(getQueryArgument(logQueryArgument));
+  const [isLoadingLog, setLoadingLog] = React.useState(true);
+  const [isLoadingFilenames, setLoadingFilenames] = React.useState(true);
+  const [error, setError] = React.useState('');
+  const [isPathOpen, setPathOpen] = React.useState(false);
+  const [isFilenameOpen, setFilenameOpen] = React.useState(false);
+  const [content, setContent] = React.useState('');
+  const [isWrapLines, setWrapLines] = useUserSettings<boolean>(
+    `${USERSETTINGS_PREFIX}.log.wrapLines`,
```

Comment:
I'd make this a shared constant somewhere so both logs always use the same value.